dali/ft/filecopy.cpp

```diff
@@ -3353,6 +3405,10 @@ void FileSprayer::spray()
     }
     addEmptyFilesToPartition();
 
+    if (!usePullOperation())
+        checkPushSupported(); // will switch to pull if push not supported
```

Comment:
This placement doesn't seem right.  Better would be for this call to be inside usePullOperation().... or in calcUsePull as suggested below

Reply:
it's too early. partitions are created later, but usePullOperation (and calcUsePull) are called early.